apps/framework-cli/src/cli.rs

```
@@ -204,15 +231,84 @@ pub async fn top_command_handler(
                 create_project_from_template(&template, name, dir_path, *no_fail_already_exists)
                     .await?;
 
-            if let Some(remote_url) = from_remote {
-                db_to_dmv2(remote_url, dir_path).await?;
-            }
+            let normalized_url = if let Some(remote_val) = from_remote {
+                // If empty or invalid, optionally prompt; otherwise validate and proceed
+                let url = if remote_val.as_deref().unwrap_or("").trim().is_empty() {
+                    use std::io::{self, Write};
+                    print!("Enter HTTPS host and port (e.g. https://your-service-id.region.clickhouse.cloud:8443)\n  ❓ Help: https://docs.fiveonefour.com/moose/getting-started/from-clickhouse#connect-to-your-remote-clickhouse\n  ☁️ ClickHouse Cloud Console: https://clickhouse.cloud/\n> ");
+                    let _ = io::stdout().flush();
+                    let mut base = String::new();
+                    io::stdin().read_line(&mut base).unwrap_or(0);
+                    let base = base.trim().trim_end_matches('/');
+
+                    print!("Enter username: ");
+                    let _ = io::stdout().flush();
+                    let mut user = String::new();
+                    io::stdin().read_line(&mut user).unwrap_or(0);
+                    let user = user.trim();
+
+                    print!("Enter password: ");
+                    let _ = io::stdout().flush();
+                    let mut pass = String::new();
+                    io::stdin().read_line(&mut pass).unwrap_or(0);
+                    let pass = pass.trim();
```

make this pattern a function
also, don't ignore the Err